roles/httpd/tasks/main.yml

```diff
@@ -0,0 +1,56 @@
+---
+- name: Install Apache httpd
+  ansible.builtin.package:
+    name:
+      - httpd
+      - mod_ssl
+    state: present
+
+- name: Set httpd_can_network_connect so Apache can connect to Puma and Gunicorn
+  ansible.posix.seboolean:
+    name: httpd_can_network_connect
+    state: true
+    persistent: true
+
+- name: Disable welcome page
+  ansible.builtin.file:
+    path: /etc/httpd/conf.d/welcome.conf
+    state: absent
+
+- name: Create cert directories
+  ansible.builtin.file:
+    path: "{{ item }}"
+    state: directory
+  loop:
+    - /etc/pki/katello/certs
+    - /etc/pki/katello/private
+
+- name: Deploy certificates
+  ansible.builtin.copy:
+    src: "{{ item.src }}"
+    dest: "/etc/pki/katello/{{ item.dest }}"
+    remote_src: true
+  loop:
+    - src: "{{ httpd_server_ca_certificate }}"
+      dest: "certs/katello-server-ca.crt"
+    - src: "{{ httpd_client_ca_certificate }}"
+      dest: "certs/katello-default-ca.crt"
+    - src: "{{ httpd_server_certificate }}"
+      dest: "certs/katello-apache.crt"
+    - src: "{{ httpd_server_key }}"
+      dest: "private/katello-apache.key"
+
+- name: Configure foreman-ssl vhost
+  ansible.builtin.template:
+    src: foreman-ssl-vhost.conf.j2
+    dest: /etc/httpd/conf.d/foreman-ssl.conf
+  vars:
+    pulp_api_backend: http://localhost:8080/
+    pulp_content_backend: http://localhost:8080/
+    foreman_backend: http://localhost:3000/
```

Comment:
Do you think we need to worry about making these variables or stick with a "static / opinionated" implementation?

Reply:
I think I'd make them variables, yes.